### PR TITLE
Adding links to roadmap and blog posts

### LINKED ISSUE #278: Feature blog posts in docs
1. Find the right places to reference recent blog posts in docs. 
2. Callout recent blogs on docs homepage.

### PR DESCRIPTION
**Pull request by jseldess**
Updating docs homepage with links to current roadmap and recent blog posts. 

Updating various pages with links to blog posts on transaction atomicity and isolation, index selection, correctness and consistency (jepson testing), SQL mapping to key-value storage, and online schema changes.

@spencerkimball, any idea where I might link to your post on atomic clocks? 

Fixes #278

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/291)

### CODE REVIEW

File: alter-table.md [hunk 1]
```diff
@@ -5,6 +5,8 @@ toc: false
 
 The `ALTER TABLE` [statement](sql-statements.html) applies a schema change to a table.
 
+{{site.data.alerts.callout_info}}For insight into how CockroachDB changes schema elements without forcing downtime, see <a href="https://www.cockroachlabs.com/blog/how-online-schema-changes-are-possible-in-cockroachdb/">Online Schema Changes in CockroachDB</a>.{{site.data.alerts.end}}
```
Comment (spencerkimball) [inline alter-table.md]: s/For insight into/To understand/
s/without forcing downtime/without requiring table locking or other user-visible downtime/

File: configure-replication-zones.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ toc: false
 
 In CockroachDB, you use **replication zones** to control the number and location of replicas for specific sets of data. Initially, there is a single, default replication zone for the entire cluster. You can adjust this default zone as well as add zones for individual databases and tables as needed. For example, you might use the default zone to replicate most data in a cluster normally within a single datacenter, while creating a specific zone to more highly replicate a certain database or table across multiple datacenters and geographies.
 
-This page explains how replication zones work and how to use the `cockroach zone` [command](cockroach-commands.html) to configure them.
+This page explains how replication zones work and how to use the `cockroach zone` [command](cockroach-commands.html) to configure them. For a detailed discussion of how CockroachDB has tested replicated data for correctness and consistency, see [DIY Jepson Testing of CockroachDB](https://www.cockroachlabs.com/blog/diy-jepsen-testing-cockroachdb/)
```
Comment (spencerkimball) [inline configure-replication-zones.md]: Hmm, not sure this is the right place for this...

File: sql-statements.md [hunk 3]
```diff
@@ -5,6 +5,8 @@ toc: false
 
 CockroachDB supports the following SQL statements. Click a statement for more details.
 
+{{site.data.alerts.callout_info}}For background on how CockroachDB maps SQL table data and table indexes to key-value storage, see <a href="https://www.cockroachlabs.com/blog/sql-in-cockroachdb-mapping-table-data-to-key-value-storage/">SQL in CockroachDB</a>.{{site.data.alerts.end}}
```
Comment (spencerkimball) [inline sql-statements.md]: This doesn't really fit here. Do we have somewhere in the FAQ where we mention that SQL is built on top of KV? That's the right place. If not, just nix this link.

File: transactions.md [hunk 4]
```diff
@@ -5,6 +5,8 @@ toc: false
 
 CockroachDB supports bundling multiple SQL statements into a single all-or-nothing transaction. Each transaction guarantees [ACID semantics](https://en.wikipedia.org/wiki/ACID) spanning arbitrary tables and rows, even when data is distributed. If a transaction succeeds, all mutations are applied together with virtual simultaneity. If any part of a transaction fails, the entire transaction is aborted, and the database is left unchanged.
 
+{{site.data.alerts.callout_info}}For a detailed discussion of CockroachDB transaction semantics by CockroachDB engineers, see <a href="https://www.cockroachlabs.com/blog/how-cockroachdb-distributes-atomic-transactions/">How CockroachDB Does Distributed Atomic Transactions</a> and <a href="https://www.cockroachlabs.com/blog/serializable-lockless-distributed-isolation-cockroachdb/">Serializable, Lockless, Distributed: Isolation in CockroachDB</a>.{{site.data.alerts.end}}
```
Comment (spencerkimball) [inline transactions.md]: s/by CockroachDB engineers//

**Comment by spencerkimball** [comment]
LGTM

File: sql-statements.md (hunk 3 above)
Comment (jseldess) [inline sql-statements.md]: Removing here. We're already linking to this post from the "Why is CockroachDB SQL?" FAQ.

File: alter-table.md (hunk 1 above)
Comment (jseldess) [inline alter-table.md]: Done.

File: transactions.md (hunk 4 above)
Comment (jseldess) [inline transactions.md]: Done.

File: configure-replication-zones.md (hunk 2 above)
Comment (jseldess) [inline configure-replication-zones.md]: Removing here. I'm linking from the consistency part of the "Does CockroachDB have ACID semantics?" FAQ and from the "How is CockroachDB strongly-consistent?" FAQ.

**Pull request merged**